nav2_docking/opennav_docking/src/docking_server.cpp

```diff
@@ -652,6 +714,13 @@ void DockingServer::undockRobot()
     geometry_msgs::msg::PoseStamped staging_pose =
       dock->getStagingPose(dock_pose.pose, dock_pose.header.frame_id);
 
+    // If we performed a rotation before docking backward, we must rotate the staging pose
```

Comment:
Sanity check for me: can you test docking / undocking 2x in a row and this all works out ok :smile:

Reply:
Of course!
I've recorded a quick video. This shows the smoking test twice in a row: docking, preemption, rotating, approaching, waiting for the charge, undocking, rotating, and so on.
[Uploading Grabación de pantalla desde 2025-05-16 09-31-54.webm…]()